sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/ReactorShim.java

```diff
@@ -69,6 +74,32 @@ public static <T> Flux<Flux<T>> windowTimeout(Flux<T> source, int maxSize, Durat
         }
     }
 
+    /**
+     * If the loaded Reactor library has disposeGracefully method then it will be called, otherwise, call dispose method.
+     *
+     * @param source scheduler.
+     * @return a {@link Mono} of scheduler call dispose or disposeGracefully.
+     */
+    public static Mono<Void> disposeGracefully(Scheduler source) {
+        if (SCHEDULER_DISPOSE_GRACEFULLY_OPERATOR_HANDLE == null) {
+            source.dispose();
+            return Mono.empty();
+        }
+        try {
+            return (Mono<Void>) SCHEDULER_DISPOSE_GRACEFULLY_OPERATOR_HANDLE.invoke(source);
+        } catch (Throwable err) {
+            // 'java.lang.invoke' throws Throwable. Given 'Error' category represents a serious
+            // abnormal thread state throw it immediately else throw via standard azure-core Logger.
+            if (err instanceof Error) {
+                throw (Error) err;
```

Comment:
Monos should return normally.  in this case, return Mono.error?

Reply:
Got it. the above method(windowTimeout) throw Error directly, so I throw Error as well.